pkg/machine/provider/platform_darwin.go

```diff
@@ -31,6 +32,9 @@ func Get() (vmconfigs.VMProvider, error) {
 		return nil, err
 	}
 
+	if resolvedVMType == define.LibKrun && runtime.GOARCH == "amd64" {
```

Comment:
This matches the type twice. The `amd64` condition could simply be moved in the switch case for libkrun which IMO would me the code better to read/understand.